Error from clang: integer value is outside the valid range of values (-Wenum-constexpr-conversion)

* System: Gentoo Linux amd64
* Compiler: clang++ 19.1.3
* dxvk version: 2.5 (2.4.1 don't have this error)
* mingw64 runtime version: 12.0.0

```
$ i686-w64-mingw32-g++ -v
clang version 19.1.3+libcxx
Target: i686-w64-windows-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm/19/bin
Configuration file: /etc/clang/cross/i686-w64-mingw32.cfg
System configuration file directory: /etc/clang
```

```
i686-w64-mingw32-g++ -Isrc/d3d11/d3d11.dll.p -Isrc/d3d11 -I../dxvk-2.5/src/d3d11 -I../dxvk-2.5/include -I../dxvk-2.5/include/vulkan/include -I../dxvk-2.5/include/spirv/include -fdiagnostics-color=always -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENI
NG_MODE_FAST -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c++17 -msse -msse2 -msse3 -mfpmath=sse -Wimplicit-fallthrough -Wno-missing-field-initializers -Wno-unused-parameter -Wno-cast-function-type -Wno-unused-private-field -Wno-mic
rosoft-exception-spec -Wno-extern-c-compat -Wno-unused-const-variable -Wno-missing-braces -DNOMINMAX -D_WIN32_WINNT=0xa00 -DDXVK_WSI_WIN32 -march=znver4 -mtune=znver4 -O3 -pipe -fomit-frame-pointer -ffunction-sections -fdata-sections -fno-plt
-mno-avx -MD -MQ src/d3d11/d3d11.dll.p/d3d11_device.cpp.obj -MF src/d3d11/d3d11.dll.p/d3d11_device.cpp.obj.d -o src/d3d11/d3d11.dll.p/d3d11_device.cpp.obj -c ../dxvk-2.5/src/d3d11/d3d11_device.cpp
In file included from ../dxvk-2.5/src/d3d11/d3d11_device.cpp:4:
In file included from ../dxvk-2.5/src/d3d11/../dxgi/dxgi_monitor.h:6:
In file included from ../dxvk-2.5/src/d3d11/../dxgi/dxgi_interfaces.h:9:
In file included from ../dxvk-2.5/src/d3d11/../dxgi/dxgi_format.h:5:
In file included from ../dxvk-2.5/src/d3d11/../dxgi/../dxvk/dxvk_device.h:4:
In file included from ../dxvk-2.5/src/d3d11/../dxgi/../dxvk/dxvk_buffer.h:9:
../dxvk-2.5/src/d3d11/../dxgi/../dxvk/dxvk_memory.h:78:3: warning: 'DxvkMemoryChunk' defined as a struct here but previously declared as a class; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
   78 |   struct DxvkMemoryChunk {
      |   ^
../dxvk-2.5/src/d3d11/../dxgi/../dxvk/dxvk_memory.h:16:3: note: did you mean struct here?
   16 |   class DxvkMemoryChunk;
      |   ^~~~~
      |   struct
In file included from ../dxvk-2.5/src/d3d11/d3d11_device.cpp:13:
In file included from ../dxvk-2.5/src/d3d11/d3d11_context_def.h:3:
In file included from ../dxvk-2.5/src/d3d11/d3d11_cmdlist.h:5:
In file included from ../dxvk-2.5/src/d3d11/d3d11_context.h:21:
../dxvk-2.5/src/d3d11/d3d11_texture.h:88:54: error: integer value 4294967295 is outside the valid range of values [0, 7] for the enumeration type 'D3D11_MAP' [-Wenum-constexpr-conversion]
   88 |     static constexpr D3D11_MAP UnmappedSubresource = D3D11_MAP(-1u);
      |                                                      ^
1 warning and 1 error generated.
```

--- a/src/d3d11/d3d11_context_imm.cpp
+++ b/src/d3d11/d3d11_context_imm.cpp
@@ -625,7 +625,7 @@ namespace dxvk {
 
     // If the texture has an image as well as a staging buffer,
     // upload the written buffer data to the image
-    bool needsUpload = mapType != D3D11_MAP_READ
+    bool needsUpload = mapType != uint32_t(D3D11_MAP_READ)
       && (mapMode == D3D11_COMMON_TEXTURE_MAP_MODE_BUFFER || mapMode == D3D11_COMMON_TEXTURE_MAP_MODE_DYNAMIC);
 
     if (needsUpload) {

--- a/src/d3d11/d3d11_device.cpp
+++ b/src/d3d11/d3d11_device.cpp
@@ -1452,10 +1452,10 @@ namespace dxvk {
      || texture->CountSubresources() <= SrcSubresource)
       return;
 
-    D3D11_MAP map = texture->GetMapType(SrcSubresource);
+    uint32_t map = texture->GetMapType(SrcSubresource);
 
-    if (map != D3D11_MAP_READ
-     && map != D3D11_MAP_READ_WRITE)
+    if (map != uint32_t(D3D11_MAP_READ)
+     && map != uint32_t(D3D11_MAP_READ_WRITE))
       return;
 
     CopySubresourceData(
@@ -1481,11 +1481,11 @@ namespace dxvk {
      || texture->CountSubresources() <= DstSubresource)
       return;
 
-    D3D11_MAP map = texture->GetMapType(DstSubresource);
+    uint32_t map = texture->GetMapType(DstSubresource);
 
-    if (map != D3D11_MAP_WRITE
-     && map != D3D11_MAP_WRITE_NO_OVERWRITE
-     && map != D3D11_MAP_READ_WRITE)
+    if (map != uint32_t(D3D11_MAP_WRITE)
+     && map != uint32_t(D3D11_MAP_WRITE_NO_OVERWRITE)
+     && map != uint32_t(D3D11_MAP_READ_WRITE))
       return;
 
     CopySubresourceData(

--- a/src/d3d11/d3d11_texture.h
+++ b/src/d3d11/d3d11_texture.h
@@ -85,7 +85,7 @@ namespace dxvk {
     
   public:
 
-    static constexpr D3D11_MAP UnmappedSubresource = D3D11_MAP(-1u);
+    static constexpr uint32_t UnmappedSubresource = ~0u;
 
     D3D11CommonTexture(
             ID3D11Resource*             pInterface,
@@ -200,10 +200,10 @@ namespace dxvk {
      * \param [in] Subresource Subresource index
      * \returns Current map mode of that subresource
      */
-    D3D11_MAP GetMapType(UINT Subresource) const {
+    uint32_t GetMapType(UINT Subresource) const {
       return Subresource < m_mapInfo.size()
-        ? D3D11_MAP(m_mapInfo[Subresource].mapType)
-        : D3D11_MAP(~0u);
+        ? m_mapInfo[Subresource].mapType
+        : UnmappedSubresource;
     }
 
     /**
@@ -216,7 +216,7 @@ namespace dxvk {
      */
     void NotifyMap(UINT Subresource, D3D11_MAP MapType) {
       if (likely(Subresource < m_mapInfo.size())) {
-        m_mapInfo[Subresource].mapType = MapType;
+        m_mapInfo[Subresource].mapType = uint32_t(MapType);
 
         if (m_mapMode == D3D11_COMMON_TEXTURE_MAP_MODE_DYNAMIC)
           CreateMappedBuffer(Subresource);
@@ -561,7 +561,7 @@ namespace dxvk {
 
     struct MappedInfo {
       D3D11_COMMON_TEXTURE_SUBRESOURCE_LAYOUT layout = { };
-      D3D11_MAP                   mapType = UnmappedSubresource;
+      uint32_t                    mapType = UnmappedSubresource;
       uint64_t                    seq     = 0u;
     };
 

--- a/src/dxvk/dxvk_memory.h
+++ b/src/dxvk/dxvk_memory.h
@@ -13,11 +13,11 @@
 namespace dxvk {
   
   class DxvkMemoryAllocator;
-  class DxvkMemoryChunk;
   class DxvkSparsePageTable;
   class DxvkSharedAllocationCache;
   class DxvkResourceAllocation;
   class DxvkPagedResource;
+  struct DxvkMemoryChunk;
 
   /**
    * \brief Memory stats

